internal/e2e/helpers.go

```diff
@@ -26,7 +26,51 @@ import (
 	"github.com/ory/keto/internal/driver"
 )
 
-func newInitializedReg(t testing.TB, dsn *dbx.DsnT, cfgOverwrites map[string]interface{}) (context.Context, driver.Registry, func(*testing.T, ...*namespace.Namespace)) {
+type namespaceTestManager struct {
+	reg     driver.Registry
+	ctx     context.Context
+	nspaces []*namespace.Namespace
+	nextID  int32
+}
+
+func (m *namespaceTestManager) add(t *testing.T, nn ...*namespace.Namespace) {
+	for _, n := range nn {
+		n.ID = m.nextID
+		m.nextID++
+		m.nspaces = append(m.nspaces, n)
+	}
+
+	require.NoError(t, m.reg.Config(m.ctx).Set(config.KeyNamespaces, m.nspaces))
+
+	t.Cleanup(func() {
+		for _, n := range nn {
+			err := errors.New("not nil")
+			var pt string
+			var ts []*relationtuple.InternalRelationTuple
+			for pt != "" || err != nil {
+				ts, pt, err = m.reg.RelationTupleManager().GetRelationTuples(m.ctx, &relationtuple.RelationQuery{
+					Namespace: n.Name,
+				}, x.WithToken(pt))
+				require.NoError(t, err)
+				require.NoError(t, m.reg.RelationTupleManager().DeleteRelationTuples(m.ctx, ts...))
+			}
+		}
+	})
+}
+
+func (m *namespaceTestManager) remove(t *testing.T, id int32) {
+	newNamespaces := make([]*namespace.Namespace, 0, len(m.nspaces))
+	for _, n := range m.nspaces {
+		if n.ID != id {
+			newNamespaces = append(newNamespaces, n)
+			break
```

Comment:
This does not seem right? You want to continue after the first one that was appended right?

```suggestion
```